x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrRetentionLeaseIT.java

```diff
@@ -324,20 +300,18 @@ public void testRetentionLeasesAreNotBeingRenewedAfterRecoveryCompletes() throws
                 assertThat(retentionLease.id(), equalTo(expectedRetentionLeaseId));
                 retentionLeases.add(currentRetentionLeases);
             }
-        });
+        }
 
         /*
          * We want to ensure that the background renewal is cancelled at the end of recovery. To do this, we will sleep a small multiple
          * of the renew interval. If the renews are not cancelled, we expect that a renewal would have been sent while we were sleeping.
          * After we wake up, it should be the case that the retention leases are the same (same timestamp) as that indicates that they were
          * not renewed while we were sleeping.
          */
-        final TimeValue renewIntervalSetting = CcrRetentionLeases.RETENTION_LEASE_RENEW_INTERVAL_SETTING.get(followerClusterSettings());
-        final long renewEnd = System.nanoTime();
-        Thread.sleep(Math.max(0, randomIntBetween(2, 4) * renewIntervalSetting.millis() - TimeUnit.NANOSECONDS.toMillis(renewEnd - start)));
+        waitForAFewRenewalIntervals(start);
 
         // now ensure that the retention leases are the same
-        assertBusy(() -> {
+        {
```

Comment:
I think it's sufficient not to do so because we are asserting that leases were _not_ renewed, so checking the primary is enough. But it does not cause any harm, so I added an explicit sync in fc4459e.